src/goto-instrument/nondet_volatile.cpp

```diff
@@ -158,6 +177,146 @@ void nondet_volatilet::nondet_volatile(
   }
 }
 
+const symbolt &nondet_volatilet::typecheck_variable(
+  const irep_idt &id,
+  const namespacet &ns)
+{
+  const symbolt *symbol;
+
+  if(ns.lookup(id, symbol))
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "given name `" + id2string(id) + "` not found in symbol table",
+      NONDET_VOLATILE_VARIABLE_OPT);
+  }
+
+  if(!symbol->is_static_lifetime || !symbol->type.get_bool(ID_C_volatile))
```

Comment:
Since the volatile variables will typically model hardware resources, it would probably make less sense to have volatile variables with automatic storage duration.